### PR TITLE
Stack merge status hooks

### PR DESCRIPTION
**Pull request by berkcaputcu**
@byroot @gmalette 

This PR adds a hook called `merge_status` that gets emitted after stack is updated. `merge_status` contains the stack parameters and last significant commit status as `merge_status` attribute.

`merge_status` can be any of these: `['locked', 'success', 'failure', 'pending']`
it falls back to the last deployed commit status when all of the undeployed commits are pending or when there is no undeployed commits.

One thing to mention is that it fires this hook after each commit to `stack`. An improvement could be that it only emits this hook after only if merge_status changes.

### CODE REVIEW

File: app/models/shipit/stack.rb [hunk 1]
```diff
@@ -116,6 +117,18 @@ def head
       commits.reachable.first.try!(:sha)
     end
 
+    def merge_status
+      if locked?
+        'locked'
+      else
+        significant_commits = commits.reachable.where('id >= ?', last_deployed_commit).order(id: :desc)
```
Comment (byroot) [inline app/models/shipit/stack.rb]: I'm not sure the last deployed commit actually matter. `id > ?` is what I'd use. So `...reacheable.newer_than(last_deployed_commit)...`

File: app/models/shipit/stack.rb [hunk 2]
```diff
@@ -116,6 +117,18 @@ def head
...
+        significant_statuses = significant_commits.map(&:significant_status)
+        return 'pending' if significant_statuses.empty? || significant_statuses.all?(&:pending?)
+
+        significant_commits.map(&:significant_status).detect { |status| !status.pending? }.simple_state
```
Comment (byroot) [inline app/models/shipit/stack.rb]: I think the code would be simple without the guard clauses:

```ruby
significant_statuses.find { |s| !s.pending? }.try!(:simple_state) || 'pending'
```

**Comment by byroot** [comment]
A couple very minor code style concerns. Otherwise LGTM.

**Comment by berkcaputcu** [comment]
I updated this with better tests and it now considers `UnknownStatus` as well.

**Comment by byroot** [comment]
:+1:

**Pull request merged**